test-utils/runtime/src/lib.rs

```diff
@@ -1360,13 +811,184 @@ fn test_witness(proof: StorageProof, root: crate::Hash) {
 	assert!(ext.storage_root(Default::default()).as_slice() != &root[..]);
 }
 
+/// Some tests require the hashed keys of the storage. As the values of hashed keys are not trivial
+/// to guess, this small module provides the values of the keys, and the code which is required to
+/// generate the keys.
+#[cfg(feature = "std")]
+pub mod storage_key_generator {
+	use super::*;
+	use sp_core::Pair;
+	use sp_keyring::AccountKeyring;
+
+	fn concat_hashes(input: &Vec<&[u8]>) -> String {
+		// sp_core::hashing::twox_128(input[0]).encode_hex::<String>()
+		input.iter().map(|s| sp_core::hashing::twox_128(s)).map(hex::encode).collect()
+	}
+
+	fn twox_64_concat(x: &[u8]) -> Vec<u8> {
+		sp_core::hashing::twox_64(x).iter().chain(x.iter()).cloned().collect::<Vec<_>>()
+	}
+
+	/// Generate the hashed keys from the raw literals
+	pub fn generate_expected_keys() -> Vec<String> {
+		let literals: Vec<&[u8]> = vec![b":code", b":extrinsic_index", b":heappages"];
+
+		let keys: Vec<Vec<&[u8]>> = vec![
+			vec![b"Babe", b"Authorities"],
+			vec![b"Babe", b"EpochConfig"],
+			vec![b"Babe", b"NextAuthorities"],
+			vec![b"Babe", b"SegmentIndex"],
+			vec![b"Babe", b":__STORAGE_VERSION__:"],
+			vec![b"Balances", b":__STORAGE_VERSION__:"],
+			vec![b"Balances", b"TotalIssuance"],
+			vec![b"SubstrateTest", b"Authorities"],
+			vec![b"SubstrateTest", b":__STORAGE_VERSION__:"],
+			vec![b"System", b"LastRuntimeUpgrade"],
+			vec![b"System", b"ParentHash"],
+			vec![b"System", b":__STORAGE_VERSION__:"],
+			vec![b"System", b"UpgradedToTripleRefCount"],
+			vec![b"System", b"UpgradedToU32RefCount"],
+		];
+
+		let mut expected_keys = keys.iter().map(concat_hashes).collect::<Vec<String>>();
+
+		expected_keys.extend(literals.iter().map(hex::encode));
+
+		let balances_map_keys = (0..16_usize)
+			.into_iter()
+			.map(|i| AccountKeyring::numeric(i).public().to_vec())
+			.chain(vec![
+				AccountKeyring::Alice.public().to_vec(),
+				AccountKeyring::Bob.public().to_vec(),
+				AccountKeyring::Charlie.public().to_vec(),
+			])
+			.map(|pubkey| {
+				sp_core::hashing::blake2_128(&pubkey)
+					.iter()
+					.chain(pubkey.iter())
+					.cloned()
+					.collect::<Vec<u8>>()
+			})
+			.map(|hash_pubkey| {
+				[concat_hashes(&vec![b"System", b"Account"]), hex::encode(hash_pubkey)].concat()
+			});
+
+		expected_keys.extend(balances_map_keys);
+
+		expected_keys.push(
+			[
+				concat_hashes(&vec![b"System", b"BlockHash"]),
+				hex::encode(0u64.using_encoded(twox_64_concat)),
+			]
+			.concat(),
+		);
+
+		expected_keys.sort();
+		expected_keys
+	}
+
+	/// Provides the hashed keys in human readable form.
```

Comment:
What of this is human readable? 

And why do we need this if we have `generate_expected_keys`?

Reply:
I found debugging [tests verifying](https://github.com/paritytech/substrate/blob/c4f425b062a749010b724a3b74e8d93b6f9b01d8/client/service/test/src/client/mod.rs#L1780-L1788) hashed keys non-trivial.

On one hand we need to know how to generate the hashes, so we need to have the code. On the other hand having the list of raw hashes together with the human-readable string indicating the origin of the hash is also useful for debugging.

Initially I had a list of hashes in [comment](https://github.com/paritytech/substrate/blob/88683c877fb757946fd7fbfd69ebb923b1335d16/client/service/test/src/client/mod.rs#L1745-L1792), but it is not maintainable too.

I was not sure what should be left in the code-base, so I decided to provide two functions: generating the hashes for the storage keys from the literals used on runtime level, and the function containing full list of hashes with documented origin.

This should make adjusting some tests easier in the future (in case when runtime is updated). However I am open to suggestions here. 

(tagging @davxy).